### ISSUE TITLE
[a11y]: Placeholder information is not announced by the screen reader appropriately.

### ISSUE THREAD

**Issue by anamhussain94** [carbon-design-system/carbon#20124]
### Package

@carbon/react

### Browser

Chrome

### Package version

v11.83.0

### Automated testing tool and ruleset

WCAG Mapping: Best Practice

### Assistive technology

JAWS

### Description

Placeholder information is not announced by the screen reader appropriately.
Expected: Screen reader should announce as 'Placeholder Enter preshared key' Type in text.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-passwordinput--overview&addons=1&panelRight=0&selectedKind=Accordion&selectedStory=Default&stories=1

### Steps to reproduce

1. Navigate to carbon/design [PasswordInput](https://react.carbondesignsystem.com/?path=/docs/components-passwordinput--overview&addons=1&panelRight=0&selectedKind=Accordion&selectedStory=Default&stories=1)
2. Use a tool of your to navigate to the field

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/4d796016-7a03-4ba8-b4d4-8153d313e700" />

The PasswordInput field has type="password". If changed to type="text" then the JAWS reads it proper.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
